### ISSUE TITLE
All time graph gives the impression of a decreased number of reports at the end

### ISSUE THREAD

**Issue by jonkri** [mysociety/fixmystreet#1956]
The All time graph on the All reports page turns downward at the end, as if the total number of reported problems got lower in late 2017. Am I correct in assuming that these lines should not be able to decrease?

![stats0](https://user-images.githubusercontent.com/777056/34680065-3368832e-f498-11e7-9a2d-8d2be9fc0a26.png)

**Comment by dracos** [comment]
It's due to the smoothing introduced to the lines, I think, the numbers haven't gone down, no! It's also because it's only the first few days of 2018 so really the 2018 dot should be much closer to the 2017 one, if you see what I mean. I guess it could extrapolate to end of current year, or something else, hmm, @zarino?

**Comment by zarino** [comment]
@dracos You’re right, the smoothing doesn’t help (making it look like a hump becoming a downward trend, rather than a level plateau, which is _actually_ what we should be seeing here). But the real issue is that, especially in the first half of a year, we’re not comparing like with like. All along the x-axis, we’ve had data points that are spaced 1 year apart, except suddenly, at the end, we’ve got a gap of maybe only a few weeks or months.

Extrapolating (so the final figure would be an _estimate_ of the total number of reports at the end of that year) would solve the line angle problem, but would be misleading when placed next to the line labels – if the "dot" at the end of the orange line is labelled "1,101,572 problems reported", but it’s only January 9th, are there 1,101,572 reports _right now_ or are we _estimating_ there’ll be 1,101,572 by the end of the year?

The proper fix would be to make sure datapoints are at the correct _horizontal_ position, rather than just being centred above their year label. This isn’t how Chart.js draws its line graphs (Chart.js just plots a bar chart and then places a dot at the top centre of each bar, linking them with a smoothed line and hiding the bars) – but maybe there’s a way we could approximate it? Or maybe there’s a better chart type (or better charting library!) we could switch to?

**Comment by dracos** [comment]
What, after all my effort making it small! ;-( As you know chart.js does have a time-based graph – http://www.chartjs.org/samples/latest/scales/time/line.html – which does look like it could put the dots correctly horizontally, but that seemed massive in size compared to just using the category x-axis, plus ISTR had issues displaying the x-axis labels? We could try and switch to the time one, but I don't think that'll be trivial.
Smoothing can be turned off with a `elements: { line: { tension: 0 } },` so at least that is easy to do as a small fix if that's enough (plus change final label to say "so far"? Might not fit in).

**Comment by dracos** [comment]
I've turned on monotone cubic interpolation which at least means it never dips down. Doesn't fix the gap being too big, but I think is good enough for the moment, and fixes this issue at least.
![screen shot 2018-01-09 at 14 56 55](https://user-images.githubusercontent.com/154364/34726937-59921a14-f54d-11e7-975c-2a1abf6110c0.png)
